packages/grid/src/Grid.tsx

```diff
@@ -175,190 +167,221 @@ const Grid = ({
-          rowHeight={ cacheRef.current.rowHeight }
-          width={ width }
-          onScroll={ handleScroll }
-          isScrollingOptOut={ isScrollingOptOut }
-          overscanColumnCount={ overscanColumnCount }
-          overscanRowCount={ overscanRowCount }
+    if (shouldFitContainer) {
+      styles.height = '100%';
+      multiGridProps.width = wrapperSize.width;
+      multiGridProps.height = wrapperSize.height;
+
+      return (
+        <ScrollSync>
+          { ({
+            onScroll,
+            scrollTop,
+          }) => (
+            <div style={ styles } ref={ wrapperRef }>
+              <MultiGrid
+                { ...multiGridProps }
+                onScroll={ onScroll }
+                scrollTop={ scrollTop }
+              />
+            </div>
+          ) }
+        </ScrollSync>
+      );
+    }
+    /* eslint-disable @typescript-eslint/no-non-null-assertion  */
+    multiGridProps.width = width!;
+    multiGridProps.height = height!;
+    /* eslint-enable @typescript-eslint/no-non-null-assertion  */
+    return (
+      <ScrollSync>
+        { ({
+          onScroll,
+          scrollTop,
+        }) => (
+          <div style={ styles } ref={ wrapperRef }>
+            <MultiGrid
+              { ...multiGridProps }
+              onScroll={ onScroll }
+              scrollTop={ scrollTop }
+            />
+          </div>
+        ) }
+      </ScrollSync>
+    );
+  }
+
+  const singleGridProps = {
+    handleSelect,
+    onChangeExpand,
+    mappedItems,
+    selectedRows,
+    cacheRef,
+    themeRef,
+    rowHeight,
+    shouldChangeColumnsWidth,
+    gridHOCMappedColumns: centerMappedColumns,
+    setGridHOCMappedColumns: setCenterMappedColumns,
+    totals,
+    onChangeColumns,
+    isScrollingOptOut,
+    overscanColumnCount,
+    overscanRowCount,
+    shouldMovingColumns,
+  };
+
+  if (shouldFitContainer) {
+    return (
+      <div ref={ wrapperRef } style={ { height: '100%' } }>
+        <InternalGrid
+          width={ wrapperSize.width }
+          height={ wrapperSize.height }
+          gridPosition="center"
+          { ...singleGridProps }
         />
-      </Body>
-      { totals && (
-        <TotalBlock
-          width={ fullWidthRef.current }
-          translateX={ scrollLeft }
-          theme={ themeRef.current }
-        >
-          { filteredColums.map((el: IColumn, index: number) => (
-            <TotalCell
-              theme={ themeRef.current }
-              width={ filteredColums.length === index + 1 ? el.width + 9 : el.width }
-            >
-              <TotalCellContent center={ !!el.center } theme={ themeRef.current }>
-                <span>{ totals[el.field] }</span>
-              </TotalCellContent>
-            </TotalCell>
-          )) }
-        </TotalBlock>
-      ) }
-    </Wrapper>
+      </div>
+    );
+  }
+
+
+  return (
+    <InternalGrid
+      /* eslint-disable @typescript-eslint/no-non-null-assertion  */
+      width={ width! }
```

Comment:
Hmm, it is strange, isn't it?
InternalGrid width shouldn't be required then?

Reply:
Yeah. I See. You have a type that says that width is required but in fact inside the component property you have with a default value. The type was written incorrectly